File: frontend/src/features/hands/lib/handSlug.ts
export const HAND_SLUG_ALPHABET = "23456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
export const HAND_SLUG_LENGTH = 10;

export function generateHandSlug(): string {
  const bytes = new Uint8Array(HAND_SLUG_LENGTH);
  crypto.getRandomValues(bytes);
  return Array.from(bytes, (byte) => HAND_SLUG_ALPHABET[byte % HAND_SLUG_ALPHABET.length]).join("");
}

export function isHandSlug(value: string): boolean {
  return (
    value.length === HAND_SLUG_LENGTH && [...value].every((ch) => HAND_SLUG_ALPHABET.includes(ch))
  );
}

File: backend/app/services/hand_slug.py
from __future__ import annotations

import secrets

HAND_SLUG_ALPHABET = "23456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz"
HAND_SLUG_LENGTH = 10


def generate_hand_slug(*, length: int = HAND_SLUG_LENGTH) -> str:
    return "".join(secrets.choice(HAND_SLUG_ALPHABET) for _ in range(length))


def is_valid_hand_slug(value: str) -> bool:
    return len(value) == HAND_SLUG_LENGTH and all(ch in HAND_SLUG_ALPHABET for ch in value)
